web/src/recent_view_ui.ts

```diff
@@ -175,9 +175,19 @@ let oldest_message_timestamp = Number.POSITIVE_INFINITY;
 function set_oldest_message_date(msg_list_data: MessageListData): void {
     const has_found_oldest = msg_list_data.fetch_status.has_found_oldest();
     const has_found_newest = msg_list_data.fetch_status.has_found_newest();
+    const oldest_message_in_data = msg_list_data.first();
+    if (oldest_message_in_data) {
+        oldest_message_timestamp = Math.min(
+            oldest_message_in_data.timestamp,
+            oldest_message_timestamp,
+        );
+    }
 
-    const first_message_timestamp = msg_list_data.first()?.timestamp ?? Number.POSITIVE_INFINITY;
-    oldest_message_timestamp = Math.min(first_message_timestamp, oldest_message_timestamp);
+    if (oldest_message_timestamp === Number.POSITIVE_INFINITY) {
+        // This was likely our first fetch which didn't result in any messages being fetched.
+        // We stay hopeful here and wait for our next fetch to update the banner.
+        return;
+    }
```

Comment:
I think the bug is that we should be looking at `first_including_muted` instead; we're guaranteed to have at least 1 message there after the first fetch unless the user has no message history at all. And it's accurate to say we're showing everything since the date of the first_including_muted message, even if there's things not displayed.

```
    first(): Message {                                                                                  
        return this._items[0];                                                                          
    }                                                                                                   
                                                                                                        
    first_including_muted(): Message {                                                                  
        return this._all_items[0];                                                                      
    }                                                                                                   
```

Reply:
Yeah, makese sense. Updated!

Reply:
I rewrote the comment and tweaked the check slightly to I think better reflect my understanding of the world. Please let me know if you see any issues with the changes.